doordeck-sdk/src/commonMain/kotlin/com/doordeck/multiplatform/sdk/internal/api/HelperClient.kt

```diff
@@ -32,45 +33,89 @@ internal open class HelperClient(
-        return if (requiresKeyRegister) {
-            // Attempt to register a key pair
-            val requiresVerification = try {
-                accountClient.registerEphemeralKeyRequest()
-                false
-            } catch (exception: LockedException) {
-                accountClient.registerEphemeralKeyWithSecondaryAuthenticationRequest()
-                true
-            }
-            AssistedLoginResponse(requiresVerification)
+        val requiresVerification = if (requiresKeyRegister) {
+            // Register the key pair
+            val assistedRegisterEphemeralKeyRequest = assistedRegisterEphemeralKeyRequest()
+            assistedRegisterEphemeralKeyRequest.requiresVerification
         } else {
-            AssistedLoginResponse(false)
+            // No key pair registration required; verification is not needed
+            false
+        }
+        return AssistedLoginResponse(requiresVerification)
+    }
+
+    /**
+     * Registers a public key for the account, using the provided key or falling back to the context manager's key if null.
+     * The method attempts registration with a simple approach first and, if it fails, retries using secondary authentication.
+     *
+     * Note: If the response indicates that verification is required
+     *  (`AssistedRegisterEphemeralKeyResponse.requiresVerification` is true),
+     *  the caller must invoke `verifyEphemeralKeyRegistration` from the account resource to complete the process.
+     */
+    suspend fun assistedRegisterEphemeralKeyRequest(publicKey: ByteArray? = null): AssistedRegisterEphemeralKeyResponse {
+        val requiresVerification = try {
+            // Attempt to register the provided or default public key
+            accountClient.registerEphemeralKeyRequest(publicKey)
+            false
+        } catch (exception: LockedException) {
+            // Retry registration using secondary authentication if the first attempt fails
+            accountClient.registerEphemeralKeyWithSecondaryAuthenticationRequest(publicKey)
+            true
         }
+        return AssistedRegisterEphemeralKeyResponse(requiresVerification)
+    }
+
+    /**
+     * Performs the standard registration process in a single function. This function performs the following steps:
+     *  * Registers a new account using the provided details.
+     *  * Generates a new key pair.
+     *  * Adds the key pair to the context manager.
+     *  * Registers the key pair for use with the account.
+     *
+     * Note: This function interacts with the context manager to store key pair data but does not manage context persistence.
+     *  Ensure the context is properly loaded and stored outside this function as required.
+     */
+    suspend fun assistedRegisterRequest(email: String, password: String, displayName: String?, force: Boolean) {
+        // Register the account with the provided details
+        accountlessClient.registrationRequest(email, password, displayName, force)
```

Comment:
It's not documented, but you can register an ephemeral key *at the same time* as registering a new user, just add an additional JSON field, `ephemeralKey`